sorts/sleep_sort.ts

```diff
@@ -35,7 +35,7 @@ export const sleepSort = async (arr: number[]): Promise<number[]> => {
 
   await Promise.all(
     arr.map(async (n: number) => {
-      await sleep(n);
+      await sleep(n*100);
```

Comment:
Why the seemingly arbitrary `*100`?

Reply:
So setting sleep for just n milliseconds would sometimes not work for some cases as it supposed to.
But due to low sleep time (n milliseconds), the first value would get resolved first before the iteration to reach the other end of an array where lower values might be present.
Let's say the array to be sorted is [1,3,5,6,0] which should result in [0,1,3,5,6]. 
In this case, 1 (index 0) will gets resolved first ( ~ 1ms) before reaching the value 0 ( last index).
Putting a multiplier gave correct results. Not sure though its the correct way of dealing with this .

Reply:
Well, there probably is no guarantee made regarding wake-up order, so sleeping is probably technically an inherently flawed way to sort an array; one could always construe a scenario where it doesn't work.

A hack of improving the likelihood of correctness like this should at the very least be explained; the "algorithm" would need at least a big fat warning that it can be arbitrarily incorrect, which IMO also makes me question whether we should add it, given that it can't guarantee reliability...

Reply:
> Well, there probably is no guarantee made regarding wake-up order, so sleeping is probably technically an inherently flawed way to sort an array; one could always construe a scenario where it doesn't work.

Agreed. Algorithm is flawed and not relevant to any actual scenario. A bit go-through resources and couldn't find its
practical usage.  (https://stackoverflow.com/questions/74917807/is-there-any-practical-use-case-for-sleep-sort)
One walkaround could be sleeping the number for (n^2) time which would avoid the edge case for very small numbers few distance apart but will get irrelevant for bigger numbers unless we can come up with something like dynamically handling sleep time according to the size of numbers. But I guess its not worth the effort.

> A hack of improving the likelihood of correctness like this should at the very least be explained; the "algorithm" would need at least a big fat warning that it can be arbitrarily incorrect, which IMO also makes me question whether we should add it, given that it can't guarantee reliability...

You are right. Its tough to prove that algorithm is reliable and can't guarantee will work in every scenario.
Its impractical but indeed an interesting concept ! which makes me think people should be aware of it.
But for that like you said, a warning could be made about its reliability and practicality and its just an another fun & creative way to sort an array.